app/lib/io/zip.js

```diff
@@ -0,0 +1,34 @@
+define([
+    'jszip'
+  , 'io/readDirRecursive'
+], function(
+    JSZip
+  , readDirRecursive
+) {
+    "use strict";
+
+    var unpack = function(zipData, io, targetPath){
+        /* TODO: Implement-me! */
+    };
+
+    var encode = function(io, sourcePath){
```

Comment:
It would be extremely cool to have an async version of this, too, I believe you have already found `obtain.js` which would be the right thing to use.

Alternativeley and to make some progress here, I propose that you just prepare your apis to be compatible with obtain in the future:

```js
// you need to import "metapolator/errors" as errors
var NotImplementedError = errors.NotImplemented;

// then
var encode = function(asyn, io, sourcePath) {
    if(async)
        throw new NotImplementedError('Asynchronous execution is not yet implemented').
```